frontend/src/components/Dashboard/ScoresTableSection/index.jsx

```diff
@@ -2,78 +2,86 @@ import React from 'react'
 import { connect } from 'react-redux'
 import { Table, TableBody, TableHeader, TableHeaderColumn, TableRow, TableRowColumn, TextField } from 'material-ui'
 
-import { fetchSubjectsList, fetchStudentsList, fetchScoresList } from 'actions/group'
+import { fetchSubjectsList, fetchStudentsList, fetchScoresList, applyNewScore } from 'actions/group'
 
 
 class ScoresTable extends React.Component {
   constructor(props) {
     super(props)
     this.state = {
-      subjectsList: null,
-      studentsList: null,
-      scoresList: null,
-      neededSubjects: null,
-      rebuiltStudentsList: null,
+      groupSubjectsList: null,
+      rebuiltSubjectsList: null,
     }
-    this.handleInput = this.handleInput.bind(this)
   }
 
-  handleInput(event){
+  handleInput(studentId, courseId, event){
+    const { applyNewScore } = this.props
     const inputValue = event.target.value
-    console.log(`Got an input: ${inputValue}`)
+    applyNewScore(studentId, courseId, inputValue)
   }
 
   componentWillMount() {
-    const { fetchSubjectsList, fetchStudentsList, fetchScoresList, groupStore } = this.props
+    const { fetchSubjectsList, fetchStudentsList, fetchScoresList } = this.props
     const subjectsPromise = new Promise((resolve, reject) => { fetchSubjectsList(resolve, reject) }),
           studentsPromise = new Promise((resolve, reject) => { fetchStudentsList(resolve, reject) }),
           scoresPromise = new Promise((resolve, reject) => { fetchScoresList(resolve, reject) })
 
     Promise.all([subjectsPromise, studentsPromise, scoresPromise]).then(() => {
-      const subjectsList = this.props.groupStore.subjects.list,
-            studentsList = this.props.groupStore.students.list,
-            scoresList = this.props.groupStore.scores.list
+      const { groupStore } = this.props
+      const subjectsList = groupStore.subjects.list,
+            studentsList = groupStore.students.list,
+            scoresList = groupStore.scores.list
+
+      // Creating a list of all subjects, formatted like {subjectId: subjectName}
+      const rebuiltSubjectsList = {}
+      subjectsList.forEach(subject => {
+        rebuiltSubjectsList[subject._id] = subject.name
+      })
 
       // Creating a list of needed subjects, formatted like {subjectId: subjectName}
-      const neededSubjects = {}
+      const groupSubjectsList = {}
       studentsList.forEach((student) => {
-        neededSubjects[student.course.subject] = subjectsList.filter(s => (s._id === student.course.subject))[0].name
+        groupSubjectsList[student.course.subject] = rebuiltSubjectsList[student.course.subject]
       })
 
       // Flattering [{name: 'foo', _id: 'bar', scoreValue: 'baz'}] to {foo: {bar: 'baz'}}
       const rebuiltStudentsList = {}
```

Comment:
You have problems with naming. First that variable actually stores not a list. Second not a list of students.
As for me that variable is just a way to have scoreValue easily extracted. So it should be named something like scores/studentsScores